modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -84,14 +87,25 @@ export class Transaction extends BaseTransaction {
    * Get the recipient account and the amount
    * transferred on this transaction
    *
-   * @returns {[string, string]} - First element is the recipient, second element is the amount
+   * @returns {Recipient} consisting
+   *  the recipient, the transfer amount, and the token name for token transfer
    */
-  private getTransferData(): [string, string] {
+  private getTransferData(): Recipient {
     let transferData;
-    this._txBody.cryptoTransfer!.transfers!.accountAmounts!.forEach((transfer) => {
+    const tokenTransfers: proto.ITokenTransferList[] = this._txBody.cryptoTransfer?.tokenTransfers || [];
+    const transfers: proto.IAccountAmount[] =
+      tokenTransfers[0]?.transfers || this._txBody.cryptoTransfer?.transfers?.accountAmounts || [];
+
+    transfers.forEach((transfer) => {
       const amount = Long.fromValue(transfer.amount!);
       if (amount.isPositive()) {
-        transferData = [stringifyAccountId(transfer.accountID!), amount.toString()];
+        transferData = {
+          address: stringifyAccountId(transfer.accountID!),
+          amount: amount.toString(),
+          ...(tokenTransfers.length && {
+            tokenName: getHbarTokenFromId(stringifyTokenId(tokenTransfers[0].token!), this._coinConfig.network)?.name,
```

Comment:
is it okay to always use `tokenTransfers[0]`?
wondering if we'll run into challenges if `tokenTransfers[0].token !== tokenTransfers[1].token`

Reply:
Yes, since we currently only support 1 token only for each transfer so tokenTransfers (splited based on each token) only have 1 element.

Reply:
Actually right, account-lib is independent from wp, so I can make to and amount used normally for now since it's not necessarily be processed in wallet-platform anyway